pypit/artrace.py

```diff
@@ -389,14 +389,18 @@ def trace_objbg_image(slf, det, sciframe, slitn, objreg, bgreg, trim=2, triml=No
     for o in range(nobj):
         wll = np.where(bgreg[0][1:, o] > bgreg[0][:-1, o])[0]
         wlr = np.where(bgreg[0][1:, o] < bgreg[0][:-1, o])[0]
+        if len(wll) < len(wlr): #< len(wlr): # JXP kludge
+            wll = np.concatenate([np.array([0]).astype(int), wll])
         # Background regions to the left of object
-        for ii in range(wll.size):
+        for ii in range(wlr.size):
             lobj = slf._lordloc[det - 1][:, slitn] + triml + wll[ii]
             robj = slf._lordloc[det - 1][:, slitn] + trimr + wlr[ii]
             rec_bg_img[:, :, o] += np.clip(spatdir - lobj.reshape(sciframe.shape[0], 1), 0.0, 1.0) - \
                                    np.clip(spatdir - robj.reshape(sciframe.shape[0], 1), 0.0, 1.0)
         wrl = np.where(bgreg[1][1:, o] > bgreg[1][:-1, o])[0]
         wrr = np.where(bgreg[1][1:, o] < bgreg[1][:-1, o])[0]
+        if len(wrr) == 0:  # JXP kludge
```

Comment:
I think a similar change is needed here as above:

```
 if len(wrr) < len(wrl):
    wrr = np.concatenate([wrr, np.array([len(bgreg[1][1:, o]) - 1]).astype(int)])
```

Reply:
ah, yes.  I had forgot to copy that in.